pkg/thanos/statefulset.go

```diff
@@ -294,7 +296,26 @@ func makeStatefulSetSpec(tr *monitoringv1.ThanosRuler, config Config, ruleConfig
 	if tr.Spec.AlertQueryURL != "" {
 		trCLIArgs = append(trCLIArgs, monitoringv1.Argument{Name: "alert.query-url", Value: tr.Spec.AlertQueryURL})
 	}
-
+	if len(tr.Spec.Version) > 0 {
+		version, err := semver.ParseTolerant(tr.Spec.Version)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to parse thanos ruler version")
+		}
+		if version.GTE(semver.MustParse("0.24.0")) {
```

Comment:
```suggestion
		if thanosVersion.GTE(semver.MustParse("0.24.0")) {
```

The `thanosVersion` variable is declared at the beginning of this function :)

Reply:
ThanosVersion is string type.